spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/DataFrameWriteTestBase.java

```diff
@@ -98,10 +100,10 @@ private Dataset<Row> createDataset(List<GenericData.Record> records, Schema sche
     }
 
     // verify that the dataframe matches
-    assertThat(rows).hasSameSizeAs(records);
-    Iterator<GenericData.Record> recordIter = records.iterator();
+    assertThat(rows.size()).isEqualTo(records.size());
```

Comment:
it is generally better to use `assertThat(rows).hasSameSizeAs(records);` as that will show you the content of rows/records when the assertion ever fails. @aihuaxu was there a particular reason why this check was changed?

Reply:
@nastra  I checked the usage and `assertThat(rows).hasSameSizeAs(records);` does show the content when the number of records doesn't match compare to isEqualTo(), helping for debug. Let me change that.